docs/design/2021-10-20-table-attributes.md

```diff
@@ -0,0 +1,108 @@
+
+```sql
+SELECT * FROM INFORMATION_SCHEMA.ATTRIBUTES;
++-------------------+-----------+----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ID                | TYPE      | ATTRIBUTES           | RANGES                                                                                                                                                                                                                                   |
++-------------------+-----------+----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| schema/test/t1/p0 | key-range | "key=value"          | [7480000000000000ff3a5f720000000000fa, 7480000000000000ff3b5f720000000000fa]                                                                                                                                                             |
+| schema/test/t1    | key-range | "key=value"          | [7480000000000000ff395f720000000000fa, 7480000000000000ff3a5f720000000000fa], [7480000000000000ff3a5f720000000000fa, 7480000000000000ff3b5f720000000000fa], [7480000000000000ff3b5f720000000000fa, 7480000000000000ff3c5f720000000000fa] |
++-------------------+-----------+----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+```
+
+### DDL Management
+Here are the concrete behaviors when doing the DDL operations with a table/partition having the attributes.
+
+#### Drop Clause 
++ For the drop table operation, the attributes will be deleted once the GC has finished
++ For the drop partition operation, the attributes will be deleted immediately
+
+#### Truncate Clause
++ For the truncate table operation, both table range and partition range of attributes will be updated.
++ For the truncate partition operation, the partition range of attributes will be updated.
+
+#### Recover/Flashback Clause
++ The attributes will be not changed once the recover/flashback process success.
+
+#### Exchange Clause
++ The attributes will be exchanged including the ID and ranges.
+
+### Privilege Management
+
+Privilege management is quite straightforward:
+
+* `ALTER TABLE` statement requires `Alter` privilege
+* `information_schema.attributes` only shows the attributes on the objects that visible to the current user
+* `ADMIN SHOW DDL` requires `Super` privilege
+
+### Inheritance
+Consider that if we add attributes for a table, all partitions of that table will inherit the attributes by default.
```

Comment:
clarify the behavior of `CREATE TABLE x LIKE y`?

Reply:
For now, `x` won't have the same attributes as `y`.

Reply:
yes i mean this is not written in the document. people shouldn't need to dig discussions around to know that.

Reply:
got